### PR TITLE
Avoid using stdin/stdout/stderr as variable names

### LINKED ISSUE #148: can not make on docker alpine
```make
In file included from /usr/local/include/postgresql/server/c.h:59,
                 from /usr/local/include/postgresql/server/postgres_fe.h:25,
                 from pgctl.c:17:
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/runprogram.h: In function 'run_program':
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/runprogram.h:75:7: error: expected identifier before '(' token
   75 |  prog.stdout = NULL;
      |       ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/runprogram.h:76:7: error: expected identifier before '(' token
   76 |  prog.stderr = NULL;
      |       ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/runprogram.h: In function 'initialize_program':
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/runprogram.h:122:7: error: expected identifier before '(' token
  122 |  prog.stdout = NULL;
      |       ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/runprogram.h:123:7: error: expected identifier before '(' token
  123 |  prog.stderr = NULL;
      |       ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/runprogram.h: In function 'free_program':
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/runprogram.h:259:12: error: expected identifier before '(' token
  259 |  if (prog->stdout != NULL)
      |            ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/runprogram.h:261:14: error: expected identifier before '(' token
  261 |   free(prog->stdout);
      |              ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/runprogram.h:264:12: error: expected identifier before '(' token
  264 |  if (prog->stderr != NULL)
      |            ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/runprogram.h:266:14: error: expected identifier before '(' token
  266 |   free(prog->stderr);
      |              ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/runprogram.h: In function 'read_from_pipes':
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/runprogram.h:378:9: error: expected identifier before '(' token
  378 |   prog->stdout = strndup(outbuf->data, outbuf->len);
      |         ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/runprogram.h:383:9: error: expected identifier before '(' token
  383 |   prog->stderr = strndup(errbuf->data, errbuf->len);
      |         ^~~~~~
pgctl.c: In function 'pg_ctl_version':
pgctl.c:87:38: error: expected identifier before '(' token
   87 |  version = parse_version_number(prog.stdout);
      |                                      ^~~~~~
pgctl.c: In function 'pg_controldata':
pgctl.c:118:12: error: expected identifier before '(' token
  118 |   if (prog.stdout == NULL)
      |            ^~~~~~
pgctl.c:128:50: error: expected identifier before '(' token
  128 |   if (!parse_controldata(&pgSetup->control, prog.stdout))
      |                                                  ^~~~~~
In file included from pgctl.c:25:
pgctl.c:131:64: error: expected identifier before '(' token
  131 |    log_warn("Failed to parse pg_controldata output:\n%s", prog.stdout);
      |                                                                ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/log/src/log.h:23:63: note: in definition of macro 'log_warn'
   23 | #define log_warn(...)  log_log(LOG_WARN,  __FILE__, __LINE__, __VA_ARGS__)
      |                                                               ^~~~~~~~~~~
In file included from /usr/local/include/postgresql/server/c.h:59,
                 from /usr/local/include/postgresql/server/postgres_fe.h:25,
                 from pgctl.c:17:
pgctl.c:144:36: error: expected identifier before '(' token
  144 |    int lineCount = splitLines(prog.stderr, errorLines, BUFSIZE);
      |                                    ^~~~~~
pgctl.c:144:20: error: too few arguments to function 'splitLines'
  144 |    int lineCount = splitLines(prog.stderr, errorLines, BUFSIZE);
      |                    ^~~~~~~~~~
In file included from pgctl.c:22:
parsing.h:41:5: note: declared here
   41 | int splitLines(char *errorMessage, char **linesArray, int size);
      |     ^~~~~~~~~~
In file included from /usr/local/include/postgresql/server/c.h:59,
                 from /usr/local/include/postgresql/server/postgres_fe.h:25,
                 from pgctl.c:17:
pgctl.c: In function 'log_program_output':
pgctl.c:581:11: error: expected identifier before '(' token
  581 |  if (prog.stdout != NULL)
      |           ^~~~~~
In file included from pgctl.c:25:
pgctl.c:583:23: error: expected identifier before '(' token
  583 |   log_info("%s", prog.stdout);
      |                       ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/log/src/log.h:22:63: note: in definition of macro 'log_info'
   22 | #define log_info(...)  log_log(LOG_INFO,  __FILE__, __LINE__, __VA_ARGS__)
      |                                                               ^~~~~~~~~~~
In file included from /usr/local/include/postgresql/server/c.h:59,
                 from /usr/local/include/postgresql/server/postgres_fe.h:25,
                 from pgctl.c:17:
pgctl.c:586:11: error: expected identifier before '(' token
  586 |  if (prog.stderr != NULL)
      |           ^~~~~~
In file included from pgctl.c:25:
pgctl.c:590:24: error: expected identifier before '(' token
  590 |    log_info("%s", prog.stderr);
      |                        ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/log/src/log.h:22:63: note: in definition of macro 'log_info'
   22 | #define log_info(...)  log_log(LOG_INFO,  __FILE__, __LINE__, __VA_ARGS__)
      |                                                               ^~~~~~~~~~~
pgctl.c:594:25: error: expected identifier before '(' token
  594 |    log_error("%s", prog.stderr);
      |                         ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/log/src/log.h:24:63: note: in definition of macro 'log_error'
   24 | #define log_error(...) log_log(LOG_ERROR, __FILE__, __LINE__, __VA_ARGS__)
      |                                                               ^~~~~~~~~~~
In file included from /usr/local/include/postgresql/server/c.h:59,
                 from /usr/local/include/postgresql/server/postgres_fe.h:25,
                 from pgctl.c:17:
pgctl.c: In function 'pg_ctl_start':
pgctl.c:725:16: error: expected identifier before '(' token
  725 |    if (program.stdout != NULL)
      |                ^~~~~~
In file included from pgctl.c:25:
pgctl.c:727:28: error: expected identifier before '(' token
  727 |     log_warn("%s", program.stdout);
      |                            ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/log/src/log.h:23:63: note: in definition of macro 'log_warn'
   23 | #define log_warn(...)  log_log(LOG_WARN,  __FILE__, __LINE__, __VA_ARGS__)
      |                                                               ^~~~~~~~~~~
In file included from /usr/local/include/postgresql/server/c.h:59,
                 from /usr/local/include/postgresql/server/postgres_fe.h:25,
                 from pgctl.c:17:
pgctl.c:741:16: error: expected identifier before '(' token
  741 |    if (program.stdout)
      |                ^~~~~~
In file included from pgctl.c:25:
pgctl.c:743:29: error: expected identifier before '(' token
  743 |     log_error("%s", program.stdout);
      |                             ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/log/src/log.h:24:63: note: in definition of macro 'log_error'
   24 | #define log_error(...) log_log(LOG_ERROR, __FILE__, __LINE__, __VA_ARGS__)
      |                                                               ^~~~~~~~~~~
In file included from /usr/local/include/postgresql/server/c.h:59,
                 from /usr/local/include/postgresql/server/postgres_fe.h:25,
                 from pgctl.c:17:
pgctl.c:758:14: error: expected identifier before '(' token
  758 |  if (program.stdout)
      |              ^~~~~~
pgctl.c:760:26: error: expected identifier before '(' token
  760 |   append_to_file(program.stdout, strlen(program.stdout), logfile);
      |                          ^~~~~~
pgctl.c:760:3: error: too few arguments to function 'append_to_file'
  760 |   append_to_file(program.stdout, strlen(program.stdout), logfile);
      |   ^~~~~~~~~~~~~~
In file included from pgctl.c:21:
file_utils.h:21:6: note: declared here
   21 | bool append_to_file(char *data, long fileSize, const char *filePath);
      |      ^~~~~~~~~~~~~~
In file included from /usr/local/include/postgresql/server/c.h:59,
                 from /usr/local/include/postgresql/server/postgres_fe.h:25,
                 from pgctl.c:17:
pgctl.c: In function 'pg_ctl_promote':
pgctl.c:905:14: error: expected identifier before '(' token
  905 |  if (program.stderr != NULL)
      |              ^~~~~~
In file included from pgctl.c:25:
pgctl.c:907:27: error: expected identifier before '(' token
  907 |   log_error("%s", program.stderr);
      |                           ^~~~~~
/home/src/pg_auto_failover/src/bin/pg_autoctl/../lib/log/src/log.h:24:63: note: in definition of macro 'log_error'
   24 | #define log_error(...) log_log(LOG_ERROR, __FILE__, __LINE__, __VA_ARGS__)
      |                                                               ^~~~~~~~~~~
make[2]: *** [<builtin>: pgctl.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/src/pg_auto_failover/src/bin/pg_autoctl'
make[1]: *** [Makefile:7: pg_autoctl] Error 2
make[1]: Leaving directory '/home/src/pg_auto_failover/src/bin'
make: *** [Makefile:37: bin] Error 2
```

### PR DESCRIPTION
**Pull request by DimCitus**
Fixes builds with the tools in the OpenBSD and Alpine Linux systems.

Fixes #95.
Fixes #118.
Fixes #148.
Fixes #153.

### CODE REVIEW

**Comment by DimCitus** [comment]
We can test the fix with the following Dockerfile to build in Alpine Linux:

```
FROM alpine:edge

RUN set -ex \
    && echo http://dl-cdn.alpinelinux.org/alpine/edge/testing >> /etc/apk/repositories \
    && apk add --no-cache --virtual .build-deps \
        clang \
        gcc \
        git \
        libedit-dev \
        libxml2-dev \
        llvm9 \
        make \
        musl-dev \
        postgresql-dev \
        zlib-dev

WORKDIR /usr/src/pg_auto_failover

COPY Makefile ./
COPY ./src/ ./src
RUN make -s clean && make -s install -j8
```

**Comment by mtuncer** [comment]
I was able to build the code under alpine:edge, but I am not yet able to run the tests with that docker file.

File: src/bin/lib/subcommands.c/runprogram.h [hunk 1]
```diff
@@ -188,17 +188,17 @@ execute_program(Program *prog)
...
-			 * We redirect /dev/null into stdin rather than closing stdin,
+			 * We redirect /dev/null into _stdin rather than closing stdin,
 			 * because apparently closing it may cause undefined behavior if
 			 * any read was to happen.
 			 */
-			int stdin = open(DEV_NULL, O_RDONLY);
+			int _stdin = open(DEV_NULL, O_RDONLY);
 
-			dup2(stdin, STDIN_FILENO);
+			dup2(_stdin, STDIN_FILENO);
```
Comment (mtuncer) [inline src/bin/lib/subcommands.c/runprogram.h]: we don't use this naming schema. Can we use a standard naming like `stdIn` as we did om`struct Program`

Comment (DimCitus) [reply src/bin/lib/subcommands.c/runprogram.h]: thanks, good point, fixed.

**Pull request merged**